ddtrace/contrib/urllib3/patch.py

```diff
@@ -0,0 +1,154 @@
+        return hostname
+
+    service_name = config.urllib3["service_name"]
+    if span._parent is not None and span._parent.service is not None:
+        service_name = span._parent.service
+    return service_name
+
+
+def _infer_argument_value(args, kwargs, pos, kw, default=None):
+    """
+    This function parses the value of a target function argument that may have been
+    passed in as a positional argument or a keyword argument. Because monkey-patched
+    functions do not define the same signature as their target function, the value of
+    arguments must be inferred from the packed args and kwargs.
+
+    Keyword arguments are prioritized, followed by the positional argument, followed
+    by the default value, if any is set.
+
+    :param args: Positional arguments
+    :param kwargs: Keyword arguments
+    :param pos: The positional index of the argument if passed in as a positional arg
+    :param kw: The name of the keyword if passed in as a keyword argument
+    :param default: The default value to return if the argument was not found in args or kwaergs
+    :return: The value of the target argument
+    """
+    if kw in kwargs:
+        return kwargs[kw]
+
+    if pos < len(args):
+        return args[pos]
+
+    return default
+
+
+def _wrap_urlopen(func, obj, args, kwargs):
+    """
+    Wrapper function for the lower-level urlopen in urllib3
+
+    :param func: The original target function "urlopen"
+    :param obj: The patched instance of ``HTTPConnectionPool``
+    :param args: Positional arguments from the target function
+    :param kwargs: Keyword arguments from the target function
+    :return: The ``HTTPResponse`` from the target function
+    """
+    request_method = _infer_argument_value(args, kwargs, 0, "method")
+    request_url = _infer_argument_value(args, kwargs, 1, "url")
+    request_headers = _infer_argument_value(args, kwargs, 3, "headers") or {}
+
+    # HTTPConnectionPool allows relative path requests; convert the request_url to an absolute url
+    if request_url.startswith("/"):
+        request_url = parse.urlunparse(
+            (
+                obj.scheme,
+                "{}:{}".format(obj.host, obj.port) if obj.port else str(obj.host),
+                request_url,
+                None,
+                None,
+                None,
+            )
+        )
+
+    # TODO before merge: abstract into utils/http.py to share with requests/aiohttp
+    parsed_uri = parse.urlparse(request_url)
+    hostname = parsed_uri.netloc
+    sanitized_url = parse.urlunparse(
+        (
+            parsed_uri.scheme,
+            parsed_uri.netloc,
+            parsed_uri.path,
+            parsed_uri.params,
+            None,  # drop parsed_uri.query
+            parsed_uri.fragment,
+        )
+    )
+
+    tracer = getattr(obj, "datadog_tracer", ddtrace.tracer)
+
+    if not tracer.enabled:
+        return func(*args, **kwargs)
+
+    with tracer.trace("urllib3.urlopen", span_type=http.TYPE) as span:
```

Comment:
The `httplib` and `requests` integrations both use span names that end in `.request`.

@justiniso Do you think it makes sense to be consistent on this naming pattern?